WCFDataService/Client/System/Data/Services/Client/QueryResult.cs

```diff
@@ -740,5 +718,20 @@ private MaterializeAtom CreateMaterializer(ProjectionPlan plan, ODataPayloadKind
                 responseMessageWrapper,
                 payloadKind);
         }
+
+        // $count queries need a way to dispose responseMessage as no enummeration occurs.
```

Comment:
No need to. The Dispose is there for requests that does not enumerate to have a way of disposing the response stream. LoadProperty enummerate and the HttpWebResponseMessage and the wrapped underlying response stream is exposed in this scenario just like when doing a regular Query. Only the /$Count Query needs the Dispose explicit.